src/components/modals/ErrorModal.vue

```diff
@@ -28,6 +29,9 @@ function close() {
                     <h5 class="title is-5">Suggestion</h5>
                     <p>{{solution}}</p>
                 </div>
+                <div class="mt-3 text-right" v-if="extraActionButton">
+                    <component :is="extraActionButton.component" v-bind="extraActionButton.props" />
```

Comment:
Based on my intuition I think it might be better to just pass around the button text and a `async () => Promise<void>` type function than a component. Not a strong opinion though.

Reply:
Added the R2ErrorExtraAction interface.
I thought that the text and the function are tightly coupled (one shouldn't exist without the other).